Orange/widgets/data/owdatasampler.py

```diff
@@ -202,6 +204,31 @@ def commit(self):
         self.send("Remaining Data", other)
 
     def updateindices(self):
+        err_msg = ""
+        repl = True
+        data_length = len(self.data)
+        num_classes = len(self.data.domain.class_var.values) \
+            if self.data.domain.has_discrete_class else 0
+
+        if self.sampling_type == self.FixedSize:
+            size = self.sampleSizeNumber
+            repl = self.replacement
+        elif self.sampling_type == self.FixedProportion:
+            size = np.ceil(self.sampleSizePercentage / 100 * data_length)
+            repl = False
+        elif data_length < self.number_of_folds:
+            err_msg = "Number of folds should not be bigger than input"
+
+        if not repl and (data_length <= size or (data_length <= num_classes
+                                                 and self.stratify)):
+            err_msg = "Output should be smaller than input"
```

Comment:
This message is confusing. It can refer to the first condition, but not to the second. Also, consider that the first condition cannot occur if sampling is `self.FixedProportion`, therefore it can be moved to `self.FixedSize`.

I think it would be better to have two different messages, something like
*"Sample size exceeds the original data size"* and
*"Not enough data for stratified sampling"*

I guess it would be better to split it into two different messages.